camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/mapred/EtlRecordReader.java

```diff
@@ -123,6 +124,8 @@ private CamusWrapper getWrappedRecord(String topicName, byte[] payload) throws I
         CamusWrapper r = null;
         try {
             r = decoder.decode(payload);
+            StatsdReporter statsd = new StatsdReporter(this.context);
+            statsd.logEventDelay(topicName, r);
```

Comment:
Never mind this whole approach just won't work till I change reporting so I just moved constructor to inside check